segmentation/include/pcl/segmentation/impl/extract_labeled_clusters.hpp

```diff
@@ -103,15 +103,29 @@ pcl::extractLabeledEuclideanClusters (const PointCloud<PointT> &cloud,
     if (seed_queue.size () >= min_pts_per_cluster && seed_queue.size () <= max_pts_per_cluster)
     {
       pcl::PointIndices r;
-      r.indices.resize (seed_queue.size ());
-      for (std::size_t j = 0; j < seed_queue.size (); ++j)
-        r.indices[j] = seed_queue[j];
-
+      r.indices = std::move(seed_queue);
+      
       std::sort (r.indices.begin (), r.indices.end ());
       r.indices.erase (std::unique (r.indices.begin (), r.indices.end ()), r.indices.end ());
 
       r.header = cloud.header;
-      labeled_clusters[cloud.points[i].label].push_back (r);   // We could avoid a copy by working directly in the vector
+      const auto label = cloud.points[i].label;
+
+      // find existing label group, if any
+      //  Map (or similar) would likely be more efficient if number of labels is high (>10-20?  needs benchmark)
+      const auto it = std::find_if( labeled_clusters.begin(), labeled_clusters.end()
+        , [label]( const std::vector<pcl::PointIndices>& pi ) {
+          return !pi.empty() && !pi.front().indices.empty() && pi.front().indices.front() == label;
+        }
+        );
+
+      if ( it == labeled_clusters.end() ) { // label group not found, add new
+        std::vector<pcl::PointIndices> group = {};
+        group.emplace_back(std::move(r));
```

Comment:
`std::vector<pcl::PointIndices> group = { std::move(r) };` reads slightly better.

Reply:
I believe that will actually result in a copy, unfortunately :(  It's disappointing that we can't move from an `initializer_list`.  Otherwise I agree that it's definitely cleaner, I'd much rather write it like that

[Link](https://cppinsights.io/lnk?code=I2luY2x1ZGUgPHZlY3Rvcj4KCnN0cnVjdCBmb28gewogICAgZm9vKCkgPSBkZWZhdWx0OwogICAgZm9vKGNvbnN0IGZvbyYpPWRlbGV0ZTsKICAgIGZvbyYgb3BlcmF0b3I9KGNvbnN0IGZvbyYpPWRlbGV0ZTsKICAgIGZvbyhmb28mJik9ZGVmYXVsdDsKICAgIGZvbyYgb3BlcmF0b3I9KGZvbyYmKT1kZWZhdWx0Owp9OyAgLy8gZm9vCgppbnQgbWFpbigpIHsKCiAgCS8vIHRoaXMgd2lsbCBjb3B5IGEgZm9vCiAgICBmb28gbXlfZm9vXzAgPSB7fTsKICAgIHN0ZDo6dmVjdG9yPGZvbz4gZm9vX3ZlY18wID0ge3N0ZDo6bW92ZShteV9mb29fMCl9OwogIAogIAkvLyB0aGlzIHdpbGwgbW92ZSBhIGZvbyBwcm9wZXJseQogICAgZm9vIG15X2Zvb18xID0ge307CiAgICBzdGQ6OnZlY3Rvcjxmb28+IGZvb192ZWNfMSA9IHt9OwogICAgZm9vX3ZlY18xLmVtcGxhY2VfYmFjayhzdGQ6Om1vdmUobXlfZm9vXzEpKTsKCn0K&insightsOptions=cpp17&std=cpp17&rev=1.0)

Reply:
Ooooh, you're totally right, I forgot that `std::initializer_list` was *that* bad and not changed because of the potential ABI break...